src/reactive-membrane.ts

```diff
@@ -76,31 +90,150 @@ const defaultValueMutated: ReactiveMembraneMutationCallback = (obj: any, key: Pr
 };
 const defaultValueDistortion: ReactiveMembraneDistortionCallback = (value: any) => value;
 
-export function wrapDescriptor(membrane: ReactiveMembrane, descriptor: PropertyDescriptor, getValue: (membrane: ReactiveMembrane, originalValue: any) => any): PropertyDescriptor {
-    const { set, get } = descriptor;
+const reserveGetterMap = new WeakMap<() => any, () => any>();
+const reverseSetterMap = new WeakMap<(v: any) => void, (v: any) => void>();
+
+export function wrapDescriptor(handler: MembraneProxyHandler, descriptor: PropertyDescriptor): PropertyDescriptor {
+    if (hasOwnProperty.call(descriptor, 'value')) {
+        descriptor.value = handler.wrapValue(descriptor.value);
+    } else {
+        const { set: originalSet, get: originalGet } = descriptor;
+        if (!isUndefined(originalGet)) {
+            const get = handler.wrapGetter(originalGet);
```

Comment:
not longer creating a new function every time... instead using the weakmap for caching and identity preservation